### ISSUE TITLE
Nullable type unions don't work as expected

### ISSUE THREAD

**Issue by thekid** [xp-framework/compiler#142]
The union types RFC specifies in its [nullable union types](https://wiki.php.net/rfc/union_types_v2#nullable_union_types) section:

> An earlier version of this RFC proposed to use ?(T1|T2) for nullable union types instead, to avoid having two ways of expressing nullability in PHP. [...] The discussion feedback was overwhelmingly in favor of supporting the T1|T2|null notation. 

XP Compiler supports both notations. However, either raise errors when compiled to PHP8.0+

* `int|string|null`: *Compile error (Type declaration 'null' must be unqualified)*
* `?(int|string)` *syntax error, unexpected token "|", expecting ";" or "{"*

**Comment by thekid** [comment]
Actually, this is not a bug! The union types RFC specifies in its [nullable union types](https://wiki.php.net/rfc/union_types_v2#nullable_union_types) section:

> An earlier version of this RFC proposed to use ?(T1|T2) for nullable union types instead, to avoid having two ways of expressing nullability in PHP. [...] The discussion feedback was overwhelmingly in favor of supporting the T1|T2|null notation.

**Comment by thekid** [comment]
However, `int|string|null` yields the following when used with XP core reflection:

```php
// PHP 8.2
lang.Nullable<?string|int>

// PHP < 8.2
lang.TypeUnion<int|string|void>
```

...which is inconsistent - the first form is what we should be getting in both cases.

**Comment by thekid** [comment]
For PHP 7, this fix is needed: https://github.com/xp-framework/core/pull/316

**Comment by thekid** [comment]
> For PHP 7, this fix is needed: [xp-framework/core#316](https://github.com/xp-framework/core/pull/316)

✅ Released in bugfix release https://github.com/xp-framework/core/releases/tag/v11.4.2

**Comment by thekid** [comment]
Fix released in https://github.com/xp-framework/compiler/releases/tag/v8.5.1
